api/v1beta1/common.go

```diff
@@ -39,11 +39,12 @@ type Destination struct {
 // with a timestamp value of when the trigger was executed.
 // E.g. vso.secrets.hashicorp.com/restartedAt: "2023-03-23T13:39:31Z"
 //
-// Supported resources: Deployment, DaemonSet, StatefulSet
+// Supported resources: Deployment, DaemonSet, StatefulSet, argo.Rollout
 type RolloutRestartTarget struct {
-	// +kubebuilder:validation:Enum={Deployment,DaemonSet,StatefulSet}
-	Kind string `json:"kind"`
-	Name string `json:"name"`
+	// +kubebuilder:validation:Enum={Deployment,DaemonSet,StatefulSet,argo.Rollout}
+	Kind       string `json:"kind"`
+	Name       string `json:"name"`
+	APIVersion string `json:"apiVersion,omitempty"`
```

Comment:
Could you elaborate on the need for the APIVersion field? It looks like there's only one version of argo's Rollout object right now.

For validation, would something like this work? Since it's not a required field:

```suggestion
        // +kubebuilder:validation:Enum={argoproj.io/v1alpha1}
	APIVersion string `json:"apiVersion,omitempty"`
```

Reply:
> Could you elaborate on the need for the APIVersion field?

I've added some comments to elaborate 
https://github.com/hashicorp/vault-secrets-operator/blob/461cf779fab39327ab4a1460236b63a3b70bfff4/internal/helpers/rollout_restart.go#L109-L110
https://github.com/hashicorp/vault-secrets-operator/blob/461cf779fab39327ab4a1460236b63a3b70bfff4/api/v1beta1/common.go#L47-L51

Reply:
> // +kubebuilder:validation:Enum={argoproj.io/v1alpha1}

I was thinking about it earlier. My concern is it doesn't validate `APIVersion` based on `Kind` (in this case argo.Rollout).

I wonder it's a good practice/benefit to add it as a first layer of validation. We have a `APIVersion` validation logic for each `Kind` in `RolloutRestartTarget()`.

Reply:
Maybe we don't need to add `APIVersion` to the CRD yet? I guess I'm having trouble understanding why someone would use it now, since there's only one argo.Rollout version served by the Argo project?

Reply:
I'm seeing benefits in both removing APIVersion (1) and keeping it (2). I'm on the fence here. Wondering if you, @benashz, and @tomhjp have any thoughts.
1. Follows a practice of only writing code that is used and simplifies CRD.
2. Prepares the code framework for a future need of concurrent RolloutRestartTargets that have different APIVersions. 

Also for my learning, when should we start incorporating a future feature framework into the current development? (i.e. how far away in the future is a feature need to be considered for starting its framework incorporation?)

Reply:
I am okay to punt on adding support for specifying the APIVersion for the target Kind. We can call YAGNI here. We should definitely leave behind some documentation that we will probably need to support APIVersion in the future.

When the APIVersion is bumped we will most definitely need it since, there can be Rollout instances of varying versions deployed to the cluster.

Reply:
I removed APIVersion and documented APIVersion consideration here
https://github.com/hashicorp/vault-secrets-operator/blob/1b79bbd54ef16af51327b7c5a8ff03f7fe78488a/api/v1beta1/common.go#L47-L50.

Let me know if I can improve it anyhow.